src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,389 @@
+import json
+import os
+from pathlib import Path
+from pickle import DEFAULT_PROTOCOL
+from typing import Any, Dict, Optional
+
+import packaging.version
+
+import toml
```

Comment:
We should likely move `toml` to `check_fastai_fastcore_pyproject_versions` which is the only place where it's used. And give a nice warning!

**Note**: You've probably identified a pattern by now. For  soft dependencies, we don't import them at the top of the file,  but where they are used, and provide clear error message for users